### PR TITLE
Rest Client Reactive - MicroProfile 2.0 features

### LINKED ISSUE #15970: Upgrade Reactive Rest Client to MP Rest Client 2.0
see https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/MicroProfile.204.2E0/near/231537705

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
fixes #15970

### CODE REVIEW

**Comment by geoand** [comment]
Can you rebase this onto `main`? There should be a CI fix there to overcome the failures you are seeing

**Comment by michalszynkiewicz** [comment]
@geoand probably not this change directly, this is for `microprofile-4` branch. I don't know what's the best course of action here. @radcortez should I cherry-pick the fix from `main`?

**Comment by geoand** [comment]
Ah, I didn't see that. In any case the commit in question is 34c17b2bafda5cf98f4550e37b3698da7e6ef1ba

**Comment by michalszynkiewicz** [comment]
I cherry-picked your change, let's see what happens :)

**Comment by geoand** [comment]
Thanks, I'll check it out later today

**Comment by michalszynkiewicz** [comment]
it worked, thanks @geoand !

File: extensions/resteasy-reactive/quarkus-jaxrs-client/runtime/src/main/java/io/quarkus/resteasy/reactive/client/runtime/ToObjectArray.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package io.quarkus.resteasy.reactive.client.runtime;
+
+import java.util.Collection;
+
+/**
+ * used by query param handling mechanism, in generated code
+ */
+@SuppressWarnings("unused")
+public class ToObjectArray {
```
Comment (geoand) [inline extensions/resteasy-reactive/quarkus-jaxrs-client/runtime/src/main/java/io/quarkus/resteasy/reactive/client/runtime/ToObjectArray.java:9]: Super minor nitpick: This should probably be `final`

**Review by geoand — COMMENTED** [review]
Is there any chance we could get some really simple tests for the new functionality in the Quarkus repo as well?

**Comment by michalszynkiewicz** [comment]
Nearly all is well covered by the MP TCK that we run. Do you think it's worth duplicating some of them?

The only thing that is not tested by the TCK is the maxRedirects setting (I exposed the vert.x client setting). If a rework is not required for other reasons, I would rather add such a test in a separate PR, together with the fix for https://github.com/quarkusio/quarkus/pull/16030/files#diff-fe05ac4b7ee5045cf73299005f3e287e821e68c27ae7afd1bcd0a03a620c43b8R56

**Comment by geoand** [comment]
Perhaps we can simply change some the existing tests to showcase some of the new functionality?

My reasoning is that sometimes failures in the TCK can easily get buried and ignored...

**Comment by radcortez** [comment]
> @geoand probably not this change directly, this is for `microprofile-4` branch. I don't know what's the best course of action here. @radcortez should I cherry-pick the fix from `main`?

I usually just rebase, but I can handle that later. We may need to push this to the new soon to be `2.x` branch.

**Comment by michalszynkiewicz** [comment]
@geoand I added a test for redirects, it tests `followRedirects` - a new MP feature, and maxRedirects - the thing I added to our client additionally

File: extensions/resteasy-reactive/quarkus-jaxrs-client/runtime/src/main/java/io/quarkus/resteasy/reactive/client/runtime/ToObjectArray.java (hunk 1 above)
Comment (michalszynkiewicz) [reply extensions/resteasy-reactive/quarkus-jaxrs-client/runtime/src/main/java/io/quarkus/resteasy/reactive/client/runtime/ToObjectArray.java:9]: forgot about it in the rework, will change in the next commit

**Review by geoand — APPROVED** [review]

**Pull request merged**